site/app/controllers/admin/ReportController.php

```diff
@@ -604,10 +607,60 @@ public function generateCustomization() {
                     $this->core->getConfig()->getCourse(),
                     $this->core->getConfig()->getSemester()
                 ),
+                'csrfToken' => $this->core->getCsrfToken(),
             ]);
         }
     }
 
+    /**
+     * @Route("/courses/{_semester}/{_course}/reports/rainbow_grades_customization/upload", methods={"POST"})
+     */
+    public function uploadRainbowConfig() {
+        $redirect_url =  $this->core->buildCourseUrl((['reports']));
+        if (empty($_FILES) || !isset($_FILES['config_upload'])) {
+            $msg = 'Upload failed: No file to upload';
+            $this->core->addErrorMessage($msg);
+            return new MultiResponse(
+                JsonResponse::getErrorResponse($msg),
+                null,
+                new RedirectResponse($redirect_url)
+            );
+        }
+
+        $upload = $_FILES['config_upload'];
+        if (!isset($upload['tmp_name']) || $upload['tmp_name'] === "") {
+            $msg = 'Upload failed: Empty tmp name for file';
+            $this->core->addErrorMessage($msg);
+            return new MultiResponse(
+                JsonResponse::getErrorResponse($msg),
+                null,
+                new RedirectResponse($redirect_url)
+            );
+        }
+
+        $rainbow_grades_dir = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "rainbow_grades");
+
+        if (!@copy($upload['tmp_name'], FileUtils::joinPaths($rainbow_grades_dir, 'customization.json'))) {
```

Comment:
Prefer to use [`move_uploaded_file`](https://www.php.net/manual/en/function.move-uploaded-file.php) for `$_FILES` uploads instead of `copy`.